### ISSUE TITLE
flutter navigator 2.0 support

### ISSUE THREAD

**Issue by pasblin** [jonataslaw/getx#1220]
**ATTENTION: DO NOT USE THIS FIELD TO ASK SUPPORT QUESTIONS. USE THE PLATFORM CHANNELS FOR THIS. THIS SPACE IS DEDICATED ONLY FOR FEATURE REQUESTS**

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Comment by mofada** [comment]
I also have this question, does getx support Navigator 2.0

**Comment by mofada** [comment]
@eduardoflorence So getx support Navigator 2.0?

**Comment by eduardoflorence** [comment]
> @eduardoflorence So getx support Navigator 2.0?

Yes, `GetMaterialApp.router()`

**Comment by mofada** [comment]
@eduardoflorence Okay thank you

**Comment by mofada** [comment]
@eduardoflorence Can you provide an examle for getx with navigator 2.0?

**Comment by tobylewis** [comment]
As far as I can see GetX supports Navigator 2.0 in as much as it has  GetMaterialApp.router() which means you can roll your own Nav 2.0 code or can use a package like Beamer.  However, it does not mean commands like Get.toNamed('/details'); work in the nav 2.0 way.  And if you are not using Get for your navigation calls, then the context-less dialogs/snackbars etc do not work.  
Now that Flutter web deployment is out of beta, support for url driven navigation is more important.  As Nav 2.0 makes the boilerplate for Nav 1.0 look trivial, it would be great if GetX offered actual commands to implement 2.0 navigation.

**Comment by SogoGolf** [comment]
We really need this too. It's it being worked on?

**Comment by GauravCalidig** [comment]
> We really need this too. It's it being worked on?

+1

**Comment by KKimj** [comment]
I am waiting to resolve this issue

**Comment by manojeeva** [comment]
I try to use Navigaiton 2.0 from the below example. It works great.
I'm planning to use it in my next project.
I found one small issue. When we put an unknown route it is not redirecting to 404. I try to put a custom unknown route in GetMaterialApp.router(), GetPage. but both are not working in root navigator and nested navigator.
I found default 404 page is used if we do not provide a custom unkown route.
If you make it work, let us know.
https://github.com/jonataslaw/getx/tree/master/example_nav2

What are your comments? Is it okay to use in production? 
Thanks
